### ISSUE TITLE
Handling of 150 GB CAT database

### ISSUE THREAD

**Issue by skrakau** [nf-core/awsmegatests#29]
Hi, for the `mag` pipeline I would like to also include the tool `CAT` in the full-size test (taxonomic classification of contigs/bins). For this, the CAT database would have to be downloaded (when the link is provided as parameter), which is `150GB` in size. @ggabernet  mentioned already this would currently cause problems.

It would be great if this can be handled in the future.

**Comment by drpatelh** [comment]
Hmmm...what if we were able to upload the database to an AWS bucket and pass that as the parameter instead? Not sure about the implications but from a simplicity perspective if the data is already on AWS will it be more reliable to obtain it that way rather than downloading it via a link?

**Comment by ggabernet** [comment]
Yes, that's probably the smartest solution. The only disadvantage is that we would be permanently storing 150GB of data, and so paying the costs for it.

We could also look into temporarily staging the file in the bucket and removing it after the pipeline has run, but it's a bit tricky to trigger the removal once the pipeline has finished.

**Comment by skrakau** [comment]
One advantage of regularly downloading it would be the additional testing of the db link, which is listed in the pipeline `docs` (as it unfortunately happens frequently that certain databases or links are suddenly not available anymore). But maybe that should be handled differently anyway.

**Comment by apeltzer** [comment]
Would be both an advantage as you said but can be quite an annoyance if its broken or failing for some reason. Saw that with the SortmeRNA databases quite often and some SILVA databases in ampliseq had similar issues :-(

**Comment by ggabernet** [comment]
If you would like to add a regular download then I think the best would be to add it as part of the pipeline. E.g. an additional optional process that downloads the database. 

The disadvantage then, though, is that for AWS we will need the instance to run this process to have available 150GB of storage. Mounting storage needs to be done in the AMI, and we use so far the same AMI for all the processes, which would mean that all processes would need 150GB of storage, which would be really expensive.

I wonder if there would be a possibility to stage directly the database to the S3 bucket as part of the pipeline, when the pipeline is run on AWS? Not so easy I think, because the AWS credentials are not passed to the processes.

**Comment by skrakau** [comment]
Is there a general recommendation that bigger input files should be downloaded in separate processes instead of using the standard nextflow pulling and caching of remote files? Because with the latter, if I see it right, the db is cached and not downloaded again even when running the pipeline without using `-resume` (which has its advantages). Or do you mean an additional download process particularly for AWS tests?

And just to know for other cases, what is currently the size limit for such files?

**Comment by ggabernet** [comment]
I had a look at how this is done in the MAG pipeline: https://github.com/nf-core/mag/blob/4fed04dee9ca377af905df9f4afadce01b93e3df/main.nf#L1358

This is what I meant with inside a process, just pulling the file as part of the pipeline. I was just confused when you mentioned that the link is in the docs, and thought maybe it is expected that people download it manually beforehand 👍 

This works only if there is enough storage in the instance. We could test it once how expensive it would be to run the pipeline with an image with 150GB + overhead storage. Otherwise the only solution I see is adding it to the S3 bucket permanently. Or as part of `iGenomes`, if you think that would be a good fit?

**Comment by skrakau** [comment]
Ah I see, thanks @ggabernet! It is not the newest CAT database, so I wouldn't necessarily add now only this to `iGenomes`.

Regarding the S3 bucket, so that sounds as if the temporary storage is not possible. Anyway, would one then overwrite the `cat_db` parameter in the `awsfulltest.yml` file, to make sure the `test_full` profile also works outside of AWS (or would this cause linting problems)?

**Comment by ggabernet** [comment]
The issue of the temporary storage, is how to trigger the download to S3 and removal after the pipeline has finished. 

One could just add the `cat_db` parameter inside the `test_full.config`, that would be cleaner I think. Or why would this not be desired?
